### ISSUE TITLE
Error when compiling with GCC 10

### ISSUE THREAD

**Issue by scaronni** [sekrit-twc/zimg#124]
A couple of errors preventing compilation with GCC 10 (Fedora 32):

```
src/zimg/colorspace/matrix3.cpp: In function 'zimg::colorspace::Vector3 zimg::colorspace::operator*(const zimg::colorspace::Vector3&, const zimg::colorspace::Vector3&)':
src/zimg/colorspace/matrix3.cpp:20:7: error: 'size_t' was not declared in this scope; did you mean 'std::size_t'?
   20 |  for (size_t i = 0; i < 3; ++i) {
      |       ^~~~~~
      |       std::size_t
In file included from /usr/include/c++/10/utility:68,
                 from /usr/include/c++/10/array:38,
                 from src/zimg/colorspace/matrix3.h:6,
                 from src/zimg/colorspace/matrix3.cpp:1:
/usr/include/c++/10/x86_64-redhat-linux/bits/c++config.h:2355:26: note: 'std::size_t' declared here
 2355 |   typedef __SIZE_TYPE__  size_t;
      |                          ^~~~~~
src/zimg/colorspace/matrix3.cpp:20:21: error: 'i' was not declared in this scope
   20 |  for (size_t i = 0; i < 3; ++i) {
      |                     ^
src/zimg/colorspace/matrix3.cpp: In function 'zimg::colorspace::Vector3 zimg::colorspace::operator*(const zimg::colorspace::Matrix3x3&, const zimg::colorspace::Vector3&)':
src/zimg/colorspace/matrix3.cpp:30:7: error: 'size_t' was not declared in this scope; did you mean 'std::size_t'?
   30 |  for (size_t i = 0; i < 3; ++i) {
      |       ^~~~~~
      |       std::size_t
In file included from /usr/include/c++/10/utility:68,
                 from /usr/include/c++/10/array:38,
                 from src/zimg/colorspace/matrix3.h:6,
                 from src/zimg/colorspace/matrix3.cpp:1:
/usr/include/c++/10/x86_64-redhat-linux/bits/c++config.h:2355:26: note: 'std::size_t' declared here
 2355 |   typedef __SIZE_TYPE__  size_t;
      |                          ^~~~~~
src/zimg/colorspace/matrix3.cpp:30:21: error: 'i' was not declared in this scope
   30 |  for (size_t i = 0; i < 3; ++i) {
      |                     ^
src/zimg/colorspace/matrix3.cpp:33:14: error: expected ';' before 'k'
   33 |   for (size_t k = 0; k < 3; ++k) {
      |              ^~
      |              ;
src/zimg/colorspace/matrix3.cpp:33:22: error: 'k' was not declared in this scope
   33 |   for (size_t k = 0; k < 3; ++k) {
      |                      ^
src/zimg/colorspace/matrix3.cpp: In function 'zimg::colorspace::Matrix3x3 zimg::colorspace::operator*(const zimg::colorspace::Matrix3x3&, const zimg::colorspace::Matrix3x3&)':
src/zimg/colorspace/matrix3.cpp:45:7: error: 'size_t' was not declared in this scope; did you mean 'std::size_t'?
   45 |  for (size_t i = 0; i < 3; ++i) {
      |       ^~~~~~
      |       std::size_t
In file included from /usr/include/c++/10/utility:68,
                 from /usr/include/c++/10/array:38,
                 from src/zimg/colorspace/matrix3.h:6,
                 from src/zimg/colorspace/matrix3.cpp:1:
/usr/include/c++/10/x86_64-redhat-linux/bits/c++config.h:2355:26: note: 'std::size_t' declared here
 2355 |   typedef __SIZE_TYPE__  size_t;
      |                          ^~~~~~
src/zimg/colorspace/matrix3.cpp:45:21: error: 'i' was not declared in this scope
   45 |  for (size_t i = 0; i < 3; ++i) {
      |                     ^
src/zimg/colorspace/matrix3.cpp:46:14: error: expected ';' before 'j'
   46 |   for (size_t j = 0; j < 3; ++j) {
      |              ^~
      |              ;
src/zimg/colorspace/matrix3.cpp:46:22: error: 'j' was not declared in this scope
   46 |   for (size_t j = 0; j < 3; ++j) {
      |                      ^
src/zimg/colorspace/matrix3.cpp:49:15: error: expected ';' before 'k'
   49 |    for (size_t k = 0; k < 3; ++k) {
      |               ^~
      |               ;
src/zimg/colorspace/matrix3.cpp:49:23: error: 'k' was not declared in this scope
   49 |    for (size_t k = 0; k < 3; ++k) {
      |                       ^
src/zimg/colorspace/matrix3.cpp: In function 'zimg::colorspace::Matrix3x3 zimg::colorspace::transpose(const zimg::colorspace::Matrix3x3&)':
src/zimg/colorspace/matrix3.cpp:101:7: error: 'size_t' was not declared in this scope; did you mean 'std::size_t'?
  101 |  for (size_t i = 0; i < 3; ++i) {
      |       ^~~~~~
      |       std::size_t
In file included from /usr/include/c++/10/utility:68,
                 from /usr/include/c++/10/array:38,
                 from src/zimg/colorspace/matrix3.h:6,
                 from src/zimg/colorspace/matrix3.cpp:1:
/usr/include/c++/10/x86_64-redhat-linux/bits/c++config.h:2355:26: note: 'std::size_t' declared here
 2355 |   typedef __SIZE_TYPE__  size_t;
      |                          ^~~~~~
src/zimg/colorspace/matrix3.cpp:101:21: error: 'i' was not declared in this scope
  101 |  for (size_t i = 0; i < 3; ++i) {
      |                     ^
src/zimg/colorspace/matrix3.cpp:102:14: error: expected ';' before 'j'
  102 |   for (size_t j = 0; j < 3; ++j) {
      |              ^~
      |              ;
src/zimg/colorspace/matrix3.cpp:102:22: error: 'j' was not declared in this scope
  102 |   for (size_t j = 0; j < 3; ++j) {
      |                      ^
make[1]: *** [Makefile:2585: src/zimg/colorspace/libzimg_internal_la-matrix3.lo] Error 1
```

Thanks.

**Comment by Kiyamou** [comment]
I can't reproduce this problem with gcc 10.0.1 (20200328) in Fedora.

I guess the problem may be related to the compilation environment.

**Comment by scaronni** [comment]
Hello, this is in a standard Fedora build root, so it's using the default (quite aggressive) compiler flags. These are as follows:

```
CXXFLAGS='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection'
```
```
LDFLAGS='-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld'
```

**Comment by sekrit-twc** [comment]
What if you compile without any flags?

**Comment by sekrit-twc** [comment]
Figured it out. This was already fixed in 9ae36d7.

**Comment by scaronni** [comment]
Thank you very much, sorry for not being reactive, having a terrible time with lockdown, work and kids at home. :face_with_head_bandage:
